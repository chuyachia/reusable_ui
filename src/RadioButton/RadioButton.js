import React from "react";
import PropTypes from "prop-types";

const RadioButton = ({
  label,
  value,
  className,
  name,
  selected,
  groupSelection,
  onChange = () => null,
  // React complain when props.checked is set without onChange function but onChange can actually be on the parent wrapper
}) => {
  var props = { name, value, onChange };
  if (selected || groupSelection === value) {
    props.checked = true;
  } else if (groupSelection !== undefined) {
    props.checked = false;
  }
  return (
    <label className={className} value={value} tabIndex={0}>
      {label}&nbsp;
      <input type="radio" {...props} />
      <span />
    </label>
  );
};

RadioButton.propTypes = {
  /** Label of the radio button intended for humans*/
  label: PropTypes.string,
  /** Classname to apply extra css */
  className: PropTypes.string,
  /** Name prop of radio button input so that only one can be selected at a time*/
  name: PropTypes.string,
  /** Value of the radio button intended for machines*/
  value: PropTypes.string,
  /** Wether the radio button is selected*/
  selected: PropTypes.bool,
  /** Value of the selected radio button in the group.
   *
   * Passed from RadioCheckGroup parent*/
  groupSelection: PropTypes.string,
  /** (event)=> null
   *
   * Function to call when the radio button changed.
   *
   * Passed from the parent if inside a RadioCheckGroup*/
  onChange: PropTypes.func,
};

export default RadioButton;
