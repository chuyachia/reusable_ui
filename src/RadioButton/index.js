import styled from "@emotion/styled";

import useTheme from "../useTheme";
import RadioButton from "./RadioButton";

export default styled(RadioButton)`
  display: ${props =>
    props.direction == "horizontal" ? "inline-block" : "block"};

  position: relative;
  cursor: pointer;
  font-family: ${props => useTheme(props, "fontSans")};
  padding-left: 1.5em;
  & input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  & span {
    position: absolute;
    top: 0;
    left: 0;
    height: 1em;
    width: 1em;
    background-color: ${props => useTheme(props, "silver")};
    border-radius: 50%;
  }
  &:hover span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondaryLight");
        case "warning":
          return useTheme(props, "warningLight");
        default:
          return useTheme(props, "primaryLight");
      }
    }};
  }
  & input:checked ~ span {
    background-color: ${props => {
      switch (props.variant) {
        case "secondary":
          return useTheme(props, "secondary");
        case "warning":
          return useTheme(props, "warning");
        default:
          return useTheme(props, "primary");
      }
    }};
  }
  span:after {
    content: "";
    position: absolute;
    display: none;
  }
  & input:checked ~ span:after {
    display: block;
  }
  & span:after {
    top: 50%;
    left: 50%;
    width: 0.6em;
    height: 0.6em;
    border-radius: 50%;
    background: ${props => useTheme(props, "silver")};
    transform: translate(-50%, -50%);
  }
`;
