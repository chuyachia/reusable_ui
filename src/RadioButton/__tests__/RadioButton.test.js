import React from "react";

import RadioButton from "../RadioButton";

describe("RadioButton", () => {
  let wrapper;
  it("should render", () => {
    wrapper = shallow(<RadioButton />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render its children with properties", () => {
    var testclassname = "testclassname";
    var testvalue = "testvalue";
    var testname = "testname";
    wrapper = shallow(
      <RadioButton className={testclassname} value={testvalue} name={testname}>
        <span>Item</span>
      </RadioButton>
    );
    expect(wrapper.find("label").prop("className")).toBe(testclassname);
    expect(wrapper.find("input").prop("name")).toBe(testname);
    expect(wrapper.find("input").prop("value")).toBe(testvalue);
  });
});
