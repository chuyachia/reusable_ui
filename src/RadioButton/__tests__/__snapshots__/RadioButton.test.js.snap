// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton should render 1`] = `
<label>
  Â 
  <input
    checked={true}
    onChange={[Function]}
    type="radio"
  />
  <span />
</label>
`;
